### DISCUSSION TITLE
cs0111 error with 11.0.0-preview6

### DISCUSSION THREAD

**Discussion by KluDev** [AvaloniaUI/Avalonia#10741] [Q&A] [answered]
**To Reproduce**

1. Create a new App dotnet new avalonia.mvvm -o DotTest -n DotTest
2. Change the PackageReference to 11.0.0-preview6 for all packages 
3. Do dotnet run
4. You will get the following error
....\DotTest\Avalonia.NameGenerator\Avalonia.NameGenerator.AvaloniaNameSourceGenerator\DotTest.Views.MainWindow.g.cs(19,21): error CS0111: Type 'MainWindow' already defines a member called 'InitializeComponent' with the same parameter types

5. Change the PackageReference to 11.0.0-preview5 for all packages
6. Do donet run
7. No error 

How can I fix this error?

**Comment by timunie** [comment, marked as answer]
@KluDev please read through the breaking changes page when updating from 0.10 ► 11.0 preview

see: https://github.com/AvaloniaUI/Avalonia/wiki/Breaking-Changes#010--110

**Reply by KluDev** [reply]
Thanks! Deleting the XamlNameReferenceGenerator from csproj solved the problem.

**Reply by jpgarza93** [reply]
Thank you, same here.

**Reply by fire-lizard** [reply]
Thank you from me too!

**Comment by DraTeots** [comment]
I also stumbled upon this problem. To anyone who get here, the relevant part of the breaking changes document: 

### XamlNameReferenceGenerator is part of the main package now
Because of that, there are two breaking changes:

- If you had `XamlNameReferenceGenerator` used in the project before, it will conflict with build in generator now. Simply delete reference to the `XamlNameReferenceGenerator` package to avoid conflicts.
- If you didn't use `XamlNameReferenceGenerator` before, now it is used automatically, which might cause compile errors (missing partial keyword, for instance). 
- If you want to disable this generator, add `<AvaloniaNameGeneratorIsEnabled>false</AvaloniaNameGeneratorIsEnabled>` to your csproj file.

So as @KluDev (and others) wrote: 

**Deleting the XamlNameReferenceGenerator from csproj solves to the problem. Going to project dependencies and deleting it with IDE GUI - works the same**
